### PR TITLE
refactor imports of optional dependencies

### LINKED ISSUE #2266: Improve Exception Handling
## 🚀 Code Quality Improvement

I came across this a few times already:
```python
try:
   # something
except Exception:
   # something
```
It is the worst possible way to handle Exceptions. It is better to catch the specific exception or at least log a message. 

### Alternatives

None. Sooner or later someone has to deal with this anyway :)

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

drop tricky try/catch. form optional imports, @awaelchli 
resolves #2266

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@awaelchli any idea how it is possible that the package is found so `XXXX_AVAILABLE` is true. but the import fails?

**Comment by Borda** [comment]
@awaelchli it seems that the conditional import does not work, it is imported even inside the if statement... :/

**Comment by akihironitta** [comment]
@Borda Seems `_module_available` needs some fix.
I ran the following script:
```python
import importlib

def _module_available(module_path: str) -> bool:
    """Testing if given module is avalaible in your env
    >>> _module_available('system')
    True
    >>> _module_available('bla.bla')
    False
    """
    mods = module_path.split('.')
    assert mods, 'nothing given to test'
    # it has to be tested as per partets
    for i in range(1, len(mods)):
        module_path = '.'.join(mods[:i])
        if importlib.util.find_spec(module_path) is None:
            return False
    return True

print(_module_available("asdf"))
print(_module_available("asdf.asdf"))
```
and encountered the following unexpected behaviour...
```txt
True
False
```

**Comment by Borda** [comment]
> @Borda Seems `_module_available` needs some fix.

Great catch, I am adding also some more test for the func...

**Comment by Borda** [comment]
@tgaddair mind have a look at why the Horovod tests are failing?

File: pytorch_lightning/utilities/__init__.py [hunk 1]
```diff
@@ -16,33 +16,39 @@
...
     return True
 
 
 APEX_AVAILABLE = _module_available("apex.amp")
-NATIVE_AMP_AVALAIBLE = hasattr(torch.cuda, "amp") and hasattr(torch.cuda.amp, "autocast")
+NATIVE_AMP_AVALAIBLE = _module_available("torch.cuda.amp")
+XLA_AVAILABLE = _module_available("torch_xla")
+HOROVOD_AVAILABLE = _module_available("horovod.torch")
+OMEGACONF_AVAILABLE = _module_available("omegaconf")
+HYDRA_AVAILABLE = _module_available("hydra")
```
Comment (Borda) [inline pytorch_lightning/utilities/__init__.py:49]: @tchaton @SeanNaren shall I make an introduction of each variable as separate PR?

**Comment by Borda** [comment]
after merging all the partial PRs this remain only for Horovod

**Review by awaelchli — APPROVED** [review]

File: tests/models/test_horovod.py [hunk 2]
```diff
@@ -155,7 +146,7 @@ def test_horovod_apex(tmpdir):
 
 @pytest.mark.skip(reason="Skip till Horovod fixes integration with Native torch.cuda.amp")
```
Comment (tchaton) [inline tests/models/test_horovod.py:147]: This test is working now I now. We can remove `@pytest.mark.skip(reason="Skip till Horovod fixes integration with Native torch.cuda.amp")`

**Review by tchaton — APPROVED** [review]

File: tests/models/test_horovod.py (hunk 2 above)
Comment (Borda) [reply tests/models/test_horovod.py:147]: seems to fail on Drone, so let it add on separate PR

**Review by SeanNaren — APPROVED** [review]

**Comment by Borda** [comment]
the TPU and GPU #4884  failing is unrelated to this PR
```
RuntimeError: Failed precondition: From /job:tpu_worker/replica:0/task:0:
2 root error(s) found.
  (0) Failed precondition: Expected comparison type FLOAT or TOTALORDER.
actual: UNSIGNED
operand: f32[32,10]

	 [[{{node XRTCompile}}]]
  (1) Failed precondition: Expected comparison type FLOAT or TOTALORDER.
actual: UNSIGNED
operand: f32[32,10]

	 [[{{node XRTCompile}}]]
	 [[XRTCompile_G6]]
```
cc: @zcain117

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LTGM !

**Pull request merged**